Source/Core/VideoCommon/OpcodeDecoding.cpp

```diff
@@ -220,18 +265,22 @@ u8* Run(DataReader src, u32* cycles, bool in_display_list)
         // load vertices
         if (src.size() < 2)
           goto end;
-        u16 num_vertices = src.Read<u16>();
-        int bytes = VertexLoaderManager::RunVertices(
-            cmd_byte & GX_VAT_MASK,  // Vertex loader index (0 - 7)
-            (cmd_byte & GX_PRIMITIVE_MASK) >> GX_PRIMITIVE_SHIFT, num_vertices, src, is_preprocess);
 
-        if (bytes < 0)
-          goto end;
+        const u16 num_vertices = src.Read<u16>();
+        if (num_vertices > 0)
+        {
+          const int vtx_attr_group = cmd_byte & GX_VAT_MASK;  // Vertex loader index (0 - 7)
+          const int primitive = (cmd_byte & GX_PRIMITIVE_MASK) >> GX_PRIMITIVE_SHIFT;
+          const int bytes = VertexLoaderManager::RunVertices(
+              vtx_attr_group, primitive, num_vertices, src, is_preprocess, totalCycles);
+          if (bytes < 0)
+            goto end;
 
-        src.Skip(bytes);
+          src.Skip(bytes);
 
-        // 4 GPU ticks per vertex, 3 CPU ticks per GPU tick
-        totalCycles += num_vertices * 4 * 3 + 6;
+          // TODO: Where are these 6 cycles coming from?
```

Comment:
The pattern throughout this switch block is 6 cycles for the initial cmd byte, and then an additional 6 cycles if the command has an argument.

By  that logic, shouldn't we be adding 12 cycles? 

Then again, I don't know if these timings were guesses, based on some measurement or based on some obscure documentation. Looks like @skidau added these timings.